awx/ui_next/src/screens/WorkflowApproval/WorkflowApprovalDetail/WorkflowApprovalDetail.jsx

```diff
@@ -0,0 +1,246 @@
+          label={i18n._(t`Explanation`)}
+          value={workflowApproval.job_explanation}
+        />
+        <Detail
+          label={i18n._(t`Workflow Job`)}
+          value={
+            sourceWorkflowJob && (
+              <Link to={`/jobs/workflow/${sourceWorkflowJob?.id}`}>
+                {`${sourceWorkflowJob?.id} - ${sourceWorkflowJob?.name}`}
+              </Link>
+            )
+          }
+        />
+        <Detail
+          label={i18n._(t`Workflow Job Template`)}
+          value={
+            sourceWorkflowJobTemplate && (
+              <Link
+                to={`/templates/workflow_job_template/${sourceWorkflowJobTemplate?.id}`}
+              >
+                {sourceWorkflowJobTemplate?.name}
+              </Link>
+            )
+          }
+        />
+        <UserDateDetail
+          label={i18n._(t`Created`)}
+          date={workflowApproval.created}
+          user={workflowApproval.summary_fields.created_by}
+        />
+        <Detail
+          label={i18n._(t`Last Modified`)}
+          value={formatDateString(workflowApproval.modified)}
+        />
+        <Detail
+          label={i18n._(t`Finished`)}
+          value={formatDateString(workflowApproval.finished)}
+        />
+        <Detail
+          label={i18n._(t`Canceled`)}
+          value={formatDateString(workflowApproval.canceled_on)}
+        />
+        <Detail
+          label={i18n._(t`Elapsed`)}
+          value={secondsToHHMMSS(workflowApproval.elapsed)}
+        />
+      </DetailList>
+      <CardActionsRow>
+        {workflowApproval.can_approve_or_deny && (
+          <>
+            <Button
+              aria-label={i18n._(t`Approve`)}
+              variant="primary"
+              onClick={approveWorkflowApproval}
+              isDisabled={isLoading}
+            >
+              {i18n._(t`Approve`)}
+            </Button>
+            <Button
+              aria-label={i18n._(t`Deny`)}
+              variant="danger"
+              onClick={denyWorkflowApproval}
+              isDisabled={isLoading}
+            >
+              {i18n._(t`Deny`)}
+            </Button>
+          </>
+        )}
+        {workflowApproval.summary_fields.user_capabilities &&
+          workflowApproval.summary_fields.user_capabilities.delete && (
+            <DeleteButton
+              name={workflowApproval.name}
+              modalTitle={i18n._(t`Delete Workflow Approval`)}
+              onConfirm={deleteWorkflowApproval}
+              isDisabled={isLoading}
+            >
+              {i18n._(t`Delete`)}
+            </DeleteButton>
+          )}
+      </CardActionsRow>
+      {deleteError && (
+        <AlertModal
+          isOpen={deleteError}
+          variant="error"
+          title={i18n._(t`Error!`)}
+          onClose={dismissDeleteError}
+        >
+          {i18n._(t`Failed to delete workflow approval.`)}
+          <ErrorDetail error={deleteError} />
+        </AlertModal>
+      )}
+      {approveError && (
```

Comment:
You could combine these 2 AlertModals, and would you also only need to have 1 useDismissableError above, which would be something like
```const {error, dismissError} = useDismissableError(approveError || denyError)```
and then the alertModals would be
```<AlertModal
          isOpen={erroor}
          variant="error"
          title={i18n._(t`Error!`)}
          onClose={dismissError}
        >
         approveError ?  {i18n._(t`Failed to approve workflow approval.`)} 
            : {i18n._(t`Failed to deny workflow approval.`)}
          <ErrorDetail error={approveError} />
        </AlertModal>
```